cmd/proxy/actions/readiness.go

```diff
@@ -0,0 +1,21 @@
+package actions
+
+import (
+	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/storage"
+	"net"
+)
+
+func getReadinessHandler(s storage.Backend) buffalo.Handler {
+	return func(c buffalo.Context) error {
+		if _, err := s.List(c, "github.com/gomods/athens"); err != nil {
+			return c.Render(500, nil)
+		}
+
+		if _, err := net.LookupIP("github.com"); err != nil {
```

Comment:
my comment got deleted:
What list does is that it checks the storage and then upstream,  so in case github is not resolvable it will fail prior to this check
https://github.com/gomods/athens/blob/b92a642f1ac49758d21781f3e27de70b2fc140c2/pkg/download/protocol.go#L62